lib/onebox/helpers.rb

```diff
@@ -113,9 +114,13 @@ def self.fetch_response(
         size_bytes = Onebox.options.max_download_kb * 1024
         http.request(request) do |response|
           if cookie = response.get_fields("set-cookie")
-            # HACK: If this breaks again in the future, use HTTP::CookieJar from gem 'http-cookie'
-            # See test: it "does not send cookies to the wrong domain"
-            redir_header = { "Cookie" => cookie.join("; ") }
+            if allow_cross_domain_cookies
+              headers["Cookie"] = cookie.join("; ")
+            else
+              # HACK: If this breaks again in the future, use HTTP::CookieJar from gem 'http-cookie'
+              # See test: it "does not send cookies to the wrong domain"
+              redir_header = { "Cookie" => cookie.join("; ") }
+            end
```

Comment:
I'm not sure I follow this change... is this necessary?

Reply:
Do you mean the existing 'hack'? I haven't looked at it much, the spec even says 'unimplemented'

https://github.com/discourse/discourse/blob/7b437c9401a52cdda06402ebd9fff0a08b67ab2a/spec/lib/onebox/helpers_spec.rb#L176-L194

If you are referring to my change to allow cross-site cookie passing, yes. It took me a while to figure out, but without passing the cookies, x.com redirects to twitter.com, and twitter.com redirects back to x.com, and so on and so forth until we hit the redirect limit